src/auto-trace.js

```diff
@@ -58,10 +64,15 @@ export function throwAsyncStacktrace(...extraMiddlewares) {
  * @param {function (Object err) => Object newErr } [extraMiddlewares] - middleware function(s) that will be called on rawError before being handled
  * @returns {Error} 
  */
-export function syncStacktrace(rawError, ...extraMiddlewares){
-	const err = rawError || new Error();
-	const middlewareErr = executeMiddleware(err, extraMiddlewares);
-	return wrapObjectWithError(middlewareErr);
+export function syncStacktrace(callback = ()=>{}) {
```

Comment:
Not sure if I like this change -- we can still call the middleware properly without doing this by doing something like this:

```js
const stacktraceErr = new Error();
const syncMiddlewareErrFunctions = executeAsyncMiddleware(stacktraceErr);
executeSyncMiddleware(syncMiddlewareErrFunctions, stacktraceErr)
return stacktraceErr;
```

So in the case of synchronous apis, the middleware is called with the same Error for both asyncErr and syncErr, but in the case of asynchronous apis, the middleware is given two separate errors. This is at least what I remember deciding on when talking with you and Bret.

What are your thoughts on this?